### PR TITLE
Improve error handling on existing seed view

### PR DESCRIPTION
**Pull request by bgptr**
This diff improves the error handling on the existing seed view by changing the input's border color to red if the seed word is not at valid position.
ksnip_20201001-200949

### CODE REVIEW

File: app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ const ExistingSeed = ({
 
   const onError = useCallback(
```
Comment (vctt94) [inline app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx:54]: Can remove this use callback

File: app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx [hunk 2]
```diff
@@ -70,21 +70,24 @@ const ExistingSeed = ({
...
-        const index = errorStr.indexOf(POSITION_ERROR);
-        const numberPosition = index + POSITION_ERROR.length + 1;
-        const endErrorStr = errorStr.slice(numberPosition + 1);
-        const beginErrorStr = errorStr.slice(0, numberPosition);
-        return beginErrorStr + (seedWord.index + 1) + endErrorStr;
-      };
-      if (seedErrorStr.includes(POSITION_ERROR)) {
-        seedErrorStr = fixPositionError(seedErrorStr, seedWord);
+      if (seedErrorStr.includes(POSITION_ERROR) && seedWord) {
+        const regexp = new RegExp(`${POSITION_ERROR} (\\d+)`, "g");
```
Comment (vctt94) [inline app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx:74]: I think adding a comment here explaining what is happening would be good

File: app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx [hunk 3]
```diff
@@ -70,21 +70,24 @@ const ExistingSeed = ({
...
+        if (regexpArray != null && typeof regexpArray[1] !== "undefined") {
+          const updatedSeedWords = [...seedWords];
+          updatedSeedWords[seedWord.index] = seedWord;
+          updatedSeedWords[regexpArray[1]] = {
+            ...updatedSeedWords[regexpArray[1]],
+            error: true
+          };
+          setSeedWords(updatedSeedWords);
+        } else {
+          console.log("invalid position error msg format");
```
Comment (vctt94) [inline app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx:85]: why this log?

**Review by vctt94 — COMMENTED** [review]
Thanks for this PR. 

I left a couple of comments, but I like the direction this is going.

**Comment by bgptr** [comment]
Thank you for your review. I fixed your comments.

**Review by vctt94 — COMMENTED** [review]
Thanks for fixing my comments.

I have a couple more comments. 

I noticed it is possible when entering a word in its wrong place, the wallet throws an error, but it does not show right after it. It only shows after you entering the next word, example:

Here the absurd word already errored, as you can see at decrediton logs:

image

But it only shows as wrong if I add the third word:

image

And if I delete the word from its position, the field border gets missed: 

image

 I also think it would be good keeping the error message instead of just making the word red, as it can be a bit confusing. Making it red and showing an error message like "word on position n is wrong" would make it clearer. What do you think?

**Comment by bgptr** [comment]
Thank you for your review. I fixed the mentioned issues and cleaned up the component a bit. Also, I added the suggested error message. It is much more straightforward now.

**Review by vctt94 — APPROVED** [review]
LGTM. Good job on this PR!

**Pull request merged**